test/extended/util/framework.go

```diff
@@ -1691,15 +1691,16 @@ type podExecutor struct {
 
 // NewPodExecutor returns an executor capable of running commands in a Pod.
 func NewPodExecutor(oc *CLI, name, image string) (*podExecutor, error) {
-	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
-	if err != nil {
-		return nil, fmt.Errorf("error: %v\n(%s)", err, out)
-	}
-	_, err = WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), ParseLabelsOrDie("name="+name), CheckPodIsReady, 1, 3*time.Minute)
-	if err != nil {
-		return nil, err
-	}
-	return &podExecutor{client: oc, podName: name}, nil
+	execPod := e2epod.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), name, func(pod *corev1.Pod) {
+		pod.Name = name
+		pod.GenerateName = ""
+		pod.Spec.Containers[0].Image = utilimage.ShellImage()
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "trap exit TERM; while true; do sleep 5; done"}
+		pod.Spec.Containers[0].Args = nil
+		pod.Spec.Containers[0].SecurityContext = e2epod.GetRestrictedContainerSecurityContext()
```

Comment:
Isn't the function defined in  https://github.com/openshift/origin/blob/91219a2661c1732509a211da4d8adf46bf40df5d/test/extended/util/pods.go#L77 already setting these pod variables, except for the TerminationMessagePolicy on line 1701?  In other words, I think your passed-in function only needs to set the TerminationMessagePolicy.  

Can you also add a comment on why TerminationMessagePolicy is needed?

Reply:
> In other words, I think your passed-in function only needs to set the TerminationMessagePolicy.

Not exactly - there are two `CreateExecPodOrFail` functions:
1. Openshift-Specific in `origin/test/extended/util/pods.go` that you pointed out
2. Upstream in `k8s.io/kubernetes/test/e2e/framework/pod/resource.go`

I'm using # 2 (Upstream `CreateExecPodOrFail`). See https://github.com/openshift/origin/pull/27635#discussion_r1072822690. I think we could use the Openshift-Specific `CreateExecPodOrFail`, in which, your point would be correct about just adding the `TerminationMessagePolicy`.

> Can you also add a comment on why TerminationMessagePolicy is needed?

Will do.

Reply:
Is not having to re-invent (or at least, re-tread) the wheel an advantage of using the openshift CreateExecPodOrFail vs. upstream CreateExecPodOrFail?

Reply:
I would say yes, my first revision of this PR used it, but was told to use the upstream function. Pinging David for clarification, he might have a good reason to not use it.